lib/doorkeeper/orm/active_record/access_grant.rb

```diff
@@ -9,12 +9,12 @@ class AccessGrant < ActiveRecord::Base
     belongs_to :application, class_name: "Doorkeeper::Application",
                              optional: true, inverse_of: :access_grants
 
-    validates :resource_owner_id,
-              :application_id,
+    validates :application_id,
               :token,
               :expires_in,
-              :redirect_uri,
               presence: true
+    validates :resource_owner_id,
```

Comment:
This change looks dangerous for me :skull: And like a breaking change...

Reply:
what is removed here - is the requirement of redirect_uri in an Access::Grant. I agree, that it could be changed to
````
validates :redirect_uri, unless: -> { respond_to?(:user_code) && !user_code.blank?}
````
So it only should be required, when it's not a device code grant.

Another change is, that resource_owner_id is only required, when there is no user_code. In any other case, there should never be a user_code. So it should not change any behavior. But maybe you see something else, it could break?